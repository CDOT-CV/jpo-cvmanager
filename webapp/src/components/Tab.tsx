import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { useLocation } from 'react-router-dom'

interface TabProps {
  activeTab: string
  label: string
  path: string

  onClick: (label: string) => void
}

const Tab = (props: TabProps) => {
  const { onClick, path, label } = props
  const location = useLocation()

  let className = 'tab-list-item'

  if (location.pathname.includes(path)) {
    className += ' tab-list-active'
  }

  return (
    <Link
      className={className}
      to={path}
      onKeyDown={(e) => {
        if (e.code === 'Space') {
          onClick(label)
        }
      }}
      onClick={() => onClick(label)}
    >
      {label}
    </Link>
  )
}

Tab.propTypes = {
  activeTab: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default Tab
