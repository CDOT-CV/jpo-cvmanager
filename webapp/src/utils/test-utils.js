/**
 * This function searches for the every react-aria SSR ids in a given HTMLElement node and replace every attribute values with a static id
 *
 * This can be usefull when you're trying to generate a snapshot of components using react-aria under the hood
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceReactAriaIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: "class",
      updateFunc: (val) =>
        val.replace(
          /css-[0-9a-z]{6}-MuiTableCell-root-MuiTablePagination-root/g,
          "css-mocked-MuiTableCell-root-MuiTablePagination-root"
        ),
    },
  ];

  container.querySelectorAll("td").forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)));
      }
    });
  });
  container.querySelectorAll("input[aria-invalid]").forEach((item) => {
    delete item["aria-invalid"];
  });
  return container;
}

export { replaceChaoticIds };
