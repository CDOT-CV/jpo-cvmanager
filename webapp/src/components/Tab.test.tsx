import React from 'react'
import { render } from '@testing-library/react'
import Tab from './Tab'
import { replaceChaoticIds } from '../utils/test-utils'
import { BrowserRouter } from 'react-router-dom'

it('should take a snapshot', () => {
  const { container } = render(
    <BrowserRouter>
      <Tab onClick={() => {}} activeTab={''} path={''} label={''} />
    </BrowserRouter>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
