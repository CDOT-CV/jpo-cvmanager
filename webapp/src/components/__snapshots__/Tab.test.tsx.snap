// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <a
    class="tab-list-item tab-list-active"
    href="/"
    tabindex="0"
  />
</div>
`;
